app/src/main/java/com/bashkevich/androidfundamentals/movieslist/viewmodel/MoviesListViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.bashkevich.androidfundamentals.movieslist.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.bashkevich.androidfundamentals.data.JsonLoad
+import com.bashkevich.androidfundamentals.data.Movie
+import kotlinx.coroutines.launch
+
+class MoviesListViewModel(
+    private val jsonLoad: JsonLoad
+) : ViewModel() {
+    private val _moviesListLiveData = MutableLiveData<List<Movie>>()
+
+    val moviesListLiveData : LiveData<List<Movie>>
+    get() = _moviesListLiveData
+
+
+
+    fun loadMoviesList(){
+        viewModelScope.launch {
```

Comment:
If you already loaded it once. Why load it again?

Reply:
do you mean check saveInstancestate?

Reply:
I mean `if (viewModel.value != null)` or even check that coroutine is still alive